bundles/org.openhab.binding.elroconnects/src/main/java/org/openhab/binding/elroconnects/internal/handler/ElroConnectsAccountHandler.java

```diff
@@ -156,71 +167,81 @@ private void stopPolling() {
-            if (devicesList != null) {
-                logger.trace("deviceList response: {}", devicesList);
-
-                List<ElroConnectsConnector> response = null;
-                try {
-                    response = gson.fromJson(devicesList, deviceListType);
-                } catch (JsonParseException parseException) {
-                    logger.warn("Parsing failed: {}", parseException.getMessage());
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            "@text/offline.request-failed");
-                    return null;
-                }
-                Map<String, ElroConnectsConnector> devices = new HashMap<>();
-                if (response != null) {
-                    response.forEach(d -> devices.put(d.getDevTid(), d));
-                }
-                this.devices = devices;
-                updateStatus(ThingStatus.ONLINE);
-            } else {
-                if (accountException == null) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+        try {
+            getControllers().handle((devicesList, accountException) -> {
+                if (devicesList != null) {
+                    logger.trace("deviceList response: {}", devicesList);
+
+                    List<ElroConnectsConnector> response = null;
+                    try {
+                        response = gson.fromJson(devicesList, deviceListType);
+                    } catch (JsonParseException parseException) {
+                        logger.warn("Parsing failed: {}", parseException.getMessage());
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                "@text/offline.request-failed");
+                        return null;
+                    }
+                    Map<String, ElroConnectsConnector> devices = new HashMap<>();
+                    if (response != null) {
+                        response.forEach(d -> devices.put(d.getDevTid(), d));
+                    }
+                    this.devices = devices;
+                    updateStatus(ThingStatus.ONLINE);
                 } else {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
-                            accountException.getLocalizedMessage());
+                    if (accountException == null) {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                    } else {
+                        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                accountException.getLocalizedMessage());
+                    }
                 }
-            }
 
-            return null;
-        });
+                return null;
+            }).get();
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE);
```

Comment:
I think you should call `Thread.currentThread().interrupt();` in this case.

Reply:
Done